boto/utils.py

```diff
@@ -1101,6 +1101,41 @@ def parse_host(hostname):
     else:
         return hostname.split(':', 1)[0]
 
+
+def get_utf8able_str(s, errors='strict'):
+    """Returns a UTF8-encodable string in PY3, UTF8 bytes in PY2.
+
+    This method is similar to six's `ensure_str()`, except it also
+    makes sure that any bytes passed in can be decoded using the
+    utf-8 codec (and raises a UnicodeDecodeError if not).
+    """
+    if six.PY2:
+        # We want to return utf-8 encoded bytes.
+        if isinstance(s, six.text_type):
+            return s.encode('utf-8', errors)
+        if isinstance(s, six.binary_type):
+            # Verify the bytes can be represented in utf-8
+            s.decode('utf-8')
+            return s
+    else:
+        # We want to return a unicode/str object.
+        if isinstance(s, six.text_type):
+            return s
+        if isinstance(s, six.binary_type):
+            s = s.decode('utf-8')
+            return s
+    raise TypeError('not expecting type "%s"' % type(s))
+
+
+def get_utf8_value(self, value):
```

Comment:
Same comment on this one -- internally, we can switch all the stuff in Boto to call `get_utf8able_str` instead.  But we don't want to change behavior for or break external users who were relying on this method and its existing behavior.

Reply:
Got that fixed! Github was giving me guff, I had to make another PR and merge it into this one's branch, but the diff should have all the feedback resolved now. :)